great_expectations/profile/basic_dataset_profiler.py

```diff
@@ -36,13 +38,21 @@ def _get_column_type(cls, df, column):
     @classmethod
     def _get_column_cardinality(cls, df, column):
 
-        num_unique = df.expect_column_unique_value_count_to_be_between(column, None, None)[
-            'result']['observed_value']
-        pct_unique = df.expect_column_proportion_of_unique_values_to_be_between(
-            column, None, None)['result']['observed_value']
+        num_unique = None
+        pct_unique = None
 
+        try:
```

Comment:
Be prepared to deal with empty columns and with exceptions in the called expectations

Reply:
Makes sense

Reply:
Instead of adding a mechanism here, I think it might be more appropriate to use the GE catch_exceptions flag (probably by setting it to on by default)